package.json

```diff
@@ -1,24 +1,36 @@
 {
   "name": "daostack-arc",
-  "version": "0.0.0-alpha.19",
+  "version": "0.0.0-alpha.25",
   "description": "A platform for building DAOs",
+  "files": [
+    "contracts/",
+    "docs/",
+    "build/contracts/",
+    "migrations/",
+    "truffle.js"
```

Comment:
please add test/

Reply:
I'm curious why "test" needs to be in the package?

Reply:
how about solium and eslinter config files?

Reply:
I'm not clear why any of that needs to be in there.  This is for people who are merely including the package (like arc-js does).  Packages aren't used for development and testing.

Could be I'm missing something....that is just in my experience.

Reply:
if images/ is not in the npm so it can be totally removed from this repo.

Reply:
> Packages aren't used for development and testing.

I guess that if we include the .sol files we should include tests + linter files.

Reply:
> if images/ is not in the npm so it can be totally removed from this repo.

do you want that included in this PR?

Reply:
why not ?

Reply:
Please add also tsconfig.json

Reply:
> why not ?

some would complain that the requested changes aren't part of the scope of this issue.  but i'm certainly that last person to so complain...  ;-)

Reply:
But I _would_ still like to understand why you are asking for the inclusion of the various entries.

Reply:
because if we are publishing code (solidity files) and its binary we also should publish tests ,compiler configuration  and env setup for these files .so any one who would like to make use of that or rebuild that can use the same setup as daostack.

Reply:
> if images/ is not in the npm so it can be totally removed from this repo.

done

Reply:
>because if we are publishing code (solidity files) and its binary we also should publish tests ,compiler configuration and env setup for these files .so any one who would like to make use of that or rebuild that can use the same setup as daostack.

In that case I believe they would obtain the repo from git, they wouldn't use an npm package.

Reply:
Corollary:   We probably should define a release in git every time we publish to NPM.

Reply:
We can enable that not just via git. easy.

Reply:
cool

Reply:
So to support this option need to get back test/ and all other env config files.
I guess it will be easy and cleaner to remove "file" in package.json and add .npmignore.

Reply:
that's weird, and not worth it if you ask me.

Reply:
This "So to support this option need to get back test/ and all other env config files." ?
or
This "I guess it will be easy and cleaner to remove "file" in package.json and add .npmignore but not so important." 

or both ?

Reply:
lol

this:

> So to support this option need to get back test/ and all other env config files.

I'm not in favor of having to do that unless there is a good reason.

Reply:
please add also contribution.md.

Reply:
'contribution.md' is also not relevant to a package, is it?  Only to the repo...

Reply:
> So to support this option need to get back test/ and all other env config files.

I am not have a very strong opinion here..but i have a feeling that if we publish code we should publish tests as well.
@leviadam what your opinion re that ?

Reply:
My position:  In my experience, packages aren't used for development and testing; they are kept minimal, just what is necessary to execute, not to test and develop.  Tests come with the repo.

Reply:
the necessary to execute (in our case deploy) are the contracts binaries.
Though this npm might be used also for dev.
I would even say that it will be nice to enable 'npm test' for this package by adding  script/test.sh which build and deploy the contracts on testrpc and run the tests on it.

Reply:
Yes, such a test script could be useful.  Just to be clear, fixing it in package.json does not imply that the test environment files need to be included in the NPM package.

Reply:
at least you need the tests ,tsconfig.json .no ?

Reply:
I can't think of any reason why we would support development via the package, nor why anyone would want to or need to.  I also can't think of any other package that does such a thing (though of course I am only familiar with a tiny percentage of npm packages).

I'm open to suggestions though...

Reply:
> at least you need the tests ,tsconfig.json .no ?

One runs tests from the repo, not the package.

Reply:
You are familiar with https://github.com/OpenZeppelin/zeppelin-solidity .right ?

Reply:
If someone ever asks for it, or we can anticipate right now why someone would be likely to need it, then sure, let's include the additional test-related stuff in the package.   Otherwise I don't recommend it.

Reply:
You know this one  i guess https://www.npmjs.com/package/zeppelin-solidity

Reply:
Indeed they do.  Any idea why?

I can find lots of counter examples.

Reply:
If you include some code (library or any other) to your product (via npm or what ever) you would like to have the ability to test it and to be sure that you are testing and using the right thing.

Reply:
>  f you include some code (library or any other) to your product (via npm or what ever) you would like to have the ability to test it and to be sure that you are testing and using the right thing.

And then we have to support them as they try to get the tools to work....

I don't know anyone who does that, and one can see that the tests passed on NPM or github.

It looks to me like most of the modules being pulled into arc right now don't include test code.

Reply:
Just to be clear:  If you are convinced this is a need and want to make it happen, I support you.  I'm just giving my position, open to change, until a decision is made.